### PR TITLE
fix(logger): overlapping log output and incorrect order of operations

### PR DESCRIPTION
**Pull request by paularmstrong**

### CODE REVIEW

**Review by FXStrato — APPROVED** [review]

File: modules/core/src/core/tasks/commands/tasks.ts [hunk 1]
```diff
@@ -262,7 +262,6 @@ function singleTaskToSpec(
 
 	let fn: PromiseFn | undefined;
 	if (cmd === '$0') {
-		logger.info([cmd, ...args]);
```
Comment (jakeleveroni) [inline modules/core/src/core/tasks/commands/tasks.ts:265]: do we not want this output anymore or is it logged elsewhere in the new impl?

File: modules/logger/src/LogStep.ts [hunk 2]
```diff
@@ -135,17 +138,34 @@ export class LogStep {
...
+	 * @internal
+	 */
+	deactivate() {
+		if (!this.#active) {
+			return;
+		}
+
+		this.#active = false;
+		if (this.#writing) {
+			this.#buffer.off('data', noop);
```
Comment (jakeleveroni) [inline modules/logger/src/LogStep.ts:163]: I meant to ask about this earlier what is this doing? looks like its unregistering a noop listener?

File: modules/logger/src/LogStep.ts [hunk 3]
```diff
@@ -198,15 +220,17 @@ export class LogStep {
 
 		// if no name, this is the root logger step
```
Comment (jakeleveroni) [inline modules/logger/src/LogStep.ts:221]: out of date comment about the logger name?

File: modules/core/src/core/tasks/commands/tasks.ts (hunk 1 above)
Comment (paularmstrong) [reply modules/core/src/core/tasks/commands/tasks.ts:265]: I added that while testing and forgot to remove it

File: modules/logger/src/LogStep.ts (hunk 2 above)
Comment (paularmstrong) [reply modules/logger/src/LogStep.ts:163]: Yeah, that's added on line 176... it prevents needing to manually call `this.#buffer.read()` to get stuff: https://nodejs.org/api/stream.html#two-reading-modes It's a bit wonky.

Comment (jakeleveroni) [reply modules/logger/src/LogStep.ts:163]: eldritch streaming knowledge got it

**Review by jakeleveroni — APPROVED** [review]

**Pull request merged**